### PR TITLE
Deny request if no conversion host is configured

### PR DESCRIPTION
**Pull request by fdupont-redhat**
This PR is the manageiq-content counterpart of https://github.com/ManageIQ/manageiq/pull/18135. It checks the request and deny it if no conversion host is available. It also sets the message of the request, so that UI can display the information.

Associated RHBZ: https://bugzilla.redhat.com/show_bug.cgi?id=1640816

### CODE REVIEW

**Comment by JPrause** [comment]
@miq-bot add_label blocker

File: content/automate/ManageIQ/Transformation/StateMachines/TransformationPlanRequestApproval.class/__methods__/deny_request.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+request = $evm.root['miq_request']
+message = $evm.object['reason']
+$evm.log('info', "Request denied because of #{message}")
```
Comment (tinaafitz) [inline content/automate/ManageIQ/Transformation/StateMachines/TransformationPlanRequestApproval.class/__methods__/deny_request.rb]: @fdupont-redhat Can you add the request ID to the log message?

File: content/automate/ManageIQ/Transformation/StateMachines/TransformationPlanRequestApproval.class/__methods__/validate_request.rb [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
-prov.source_vms.each { |vm| prov.approve_vm(vm) if prov.validate_vm(vm) }
+request = $evm.root['miq_request']
+
+unless request.validate_conversion_hosts
+  $evm.object['reason'] = 'No conversion host configured'
+  request.message = 'No conversion host configured'
+  exit MIQ_ABORT
+end
+
+request.source_vms.each { |vm| request.approve_vm(vm) if request.validate_vm(vm) }
```
Comment (gmcculloug) [inline content/automate/ManageIQ/Transformation/StateMachines/TransformationPlanRequestApproval.class/__methods__/validate_request.rb]: This should eventually move to the backend, but not in this PR.

File: content/automate/ManageIQ/Transformation/StateMachines/TransformationPlanRequestApproval.class/__methods__/validate_request.rb [hunk 3]
```diff
@@ -1,5 +1,9 @@
...
-# Description: Placeholder for service request validation
-#
-prov = $evm.root['miq_request']
-prov.source_vms.each { |vm| prov.approve_vm(vm) if prov.validate_vm(vm) }
+request = $evm.root['miq_request']
+
+unless request.validate_conversion_hosts
+  $evm.object['reason'] = 'No conversion host configured'
+  request.message = 'No conversion host configured'
+  exit MIQ_ABORT
```
Comment (gmcculloug) [inline content/automate/ManageIQ/Transformation/StateMachines/TransformationPlanRequestApproval.class/__methods__/validate_request.rb]: @tinaafitz @mkanoor Should this remain `exit` or is it better to use `raise`?

Comment (mkanoor) [reply content/automate/ManageIQ/Transformation/StateMachines/TransformationPlanRequestApproval.class/__methods__/validate_request.rb]: @gmcculloug It doesn't hurt to have the exit there, we solved the spec problem with the approach listed here
https://github.com/ManageIQ/manageiq-content/blob/166a3df922e31953f63db76dd644df734fd1a40d/spec/content/automate/ManageIQ/System/CommonMethods/Utils.class/__methods__/log_object_spec.rb#L81

Which allows us to test different exit codes. So if there are specs for this change it should follow a similar pattern if the method is being tested directly outside of the DRb and Automate Engine framework.

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 82ef82a33f04bdb140115024c8467d3492401180
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Mon Nov 5 16:16:42 2018 -0500

    Merge pull request #455 from fdupont-redhat/bz1640816
    
    Deny request if no conversion host is configured
    
    (cherry picked from commit 4d84575e1512153dee113fe1db14ebf6d9b72cc9)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1640816
```

**Pull request merged**